packages/gatsby/src/utils/cache.ts

```diff
@@ -1,58 +1,75 @@
-const fs = require(`fs-extra`)
-const manager = require(`cache-manager`)
-const fsStore = require(`cache-manager-fs-hash`)
-const path = require(`path`)
+import manager, { Store, StoreConfig, CachingConfig } from "cache-manager"
+import fs from "fs-extra"
+import fsStore from "cache-manager-fs-hash"
+import path from "path"
 
 const MAX_CACHE_SIZE = 250
 const TTL = Number.MAX_SAFE_INTEGER
 
-class Cache {
-  constructor({ name = `db`, store = fsStore } = {}) {
+interface ICacheProperties {
+  name?: string
+  store?: Store
+}
+
+export default class Cache {
+  public name: string
+  public store: Store
+  public cache?: manager.Cache
+
+  constructor({ name = `db`, store = fsStore }: ICacheProperties = {}) {
     this.name = name
     this.store = store
   }
 
-  get directory() {
+  get directory(): string {
     return path.join(process.cwd(), `.cache/caches/${this.name}`)
   }
 
-  init() {
+  init(): Cache {
     fs.ensureDirSync(this.directory)
 
-    const caches = [
+    const configs: StoreConfig[] = [
       {
         store: `memory`,
         max: MAX_CACHE_SIZE,
+        ttl: TTL,
       },
       {
         store: this.store,
+        ttl: TTL,
         options: {
           path: this.directory,
           ttl: TTL,
         },
       },
-    ].map(cache => manager.caching(cache))
+    ]
+
+    const caches = configs.map(cache => manager.caching(cache))
 
     this.cache = manager.multiCaching(caches)
 
     return this
   }
 
-  get(key) {
+  get<T = unknown>(key): Promise<T | undefined> {
     return new Promise(resolve => {
-      this.cache.get(key, (err, res) => {
+      // eslint-disable-next-line no-unused-expressions
+      this.cache?.get<T>(key, (err, res) => {
         resolve(err ? undefined : res)
       })
     })
   }
 
-  set(key, value, args = {}) {
+  set<T>(
+    key: string,
+    value: T,
+    args: CachingConfig = { ttl: TTL }
+  ): Promise<T | undefined> {
     return new Promise(resolve => {
-      this.cache.set(key, value, args, err => {
+      // eslint-disable-next-line no-unused-expressions
```

Comment:
Is eslint complaining about the potential for `this.cache` to be undefined or smth?

Reply:
I think our version of eslint doesn;t recognise optional chaining, so it thought it was an unused statement

Reply:
Let's look into updating. We should be able to easily use the nullish op